spec/actors/curation_concerns/file_set_actor_spec.rb

```diff
@@ -151,21 +151,26 @@
       expect { file_set.reload }.to raise_error ActiveFedora::ObjectNotFoundError
     end
 
-    context "representative of a work" do
+    context "representative and thumbnail of a work" do
       let!(:work) do
         work = create(:generic_work)
         # this is not part of a block on the create, since the work must be saved
         # before the representative can be assigned
         work.ordered_members << file_set
         work.representative = file_set
+        work.thumbnail = file_set
         work.save
         work
       end
 
-      it "removes representative and the proxy association" do
-        expect(work.reload.representative_id).to eq(file_set.id)
+      it "removes representative, thumbnail, and the proxy association" do
+        gw = GenericWork.load_instance_from_solr(work.id)
+        expect(gw.representative_id).to eq(file_set.id)
+        expect(gw.thumbnail_id).to eq(file_set.id)
         expect { actor.destroy }.to change { ActiveFedora::Aggregation::Proxy.count }.by(-1)
-        expect(work.reload.representative_id).to be_nil
```

Comment:
@jcoyne This is what is tricky about this code. 

Before the changes in this PR accessing the `work` from Fedora after deleting the `file_set` gave the correct result (i.e. `gw.thumbnail = nil`) However, accessing the `work` from Solr gave the wrong results (i.e. `gw.thumbnail_id = "old file set id"`)

I am not sure if we should add a ticket to ActiveFedora::Aggregation or ActiveFedora::Aggregation::Proxy to handle this at a lower level of the stack. 

For now this PR takes care of the issue.